### DISCUSSION TITLE
Relative Dates Not Working

### DISCUSSION THREAD

**Discussion by ethan-atrium-ai** [jamessimone/apex-rollup#224] [Q&A]
We installed the latest version of apex-rollup for testing. We are attempting to use this WHERE clause 
Opportunity.CreatedDate = LAST_N_DAYS: 2 - this doesn't seem to correctly filter. It rolls up all opportunities to the Number field we are using as the target field even though we only have 2 opps within the last 2 days. I tried the below as well

Opportunity.CreatedDate = TODAY - this gives an error 'Invalid conversion from runtime type String to Datetime'

I am not sure if we are incorrectly entering the WHERE clause or missing configuration, but we haven't been able to successfully rollup any values at this point using relative date literals

**Comment by jamessimone** [comment]
Hey @ethan-atrium-ai - is the calculation item an opportunity? If so, the where clause should just be `CreatedDate = LAST_N_DAYS: 2`.

**Reply by ethan-atrium-ai** [reply]
Thanks @jamessimone ! I tried the below just now and I have 3 opportunities created today and yesterday. It only rolls up 1 to the Account rollup. 
CreatedDate = LAST_N_DAYS:2

When I try the below I expect 2 but the rollup shows 0. I am running the calculation from the LWC screen. Creating a new opportunity still results in 0. I have the trigger code added on the opportunity object.
CreatedDate = TODAY

**Reply by ethan-atrium-ai** [reply]
@jamessimone Looks like the = doesn't work as I was expecting for the relative date literals. I needed to include > or < as well to get values in the rollups.

**Reply by jamessimone** [reply]
@ethan-atrium-ai that's odd though, I have tests to support `=` with the ranged date literals, so I would expect them to work. On the one hand, I'm glad you managed to quickly find a work-around - on the other, I need to ensure the date literal stuff is completely solid (as it's a relatively new feature), so I will have to get a reproduction of the issue you're describing locally!

**Reply by ethan-atrium-ai** [reply]
Thanks for the quick responses @jamessimone ! It could definitely be user error. I'm still playing with it, but so far it seems including the > or < gets me what I need. Thanks again for your work on this!

**Reply by jamessimone** [reply]
You're very welcome! I will keep you updated here if I found out anything related to what you reported; I was actually working on the date literals earlier today, anyway, because I never finished adding support for SOQL date functions (like `CALENDAR_YEAR()`), so now's the _best_ time to be debugging this particular area of the codebase!
